sdk/core/azure-core-serializer-json-jackson/src/main/java/com/azure/core/serializer/json/jackson/JacksonPropertyNameSerializer.java

```diff
@@ -0,0 +1,41 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+package com.azure.core.serializer.json.jackson;
+
+import com.azure.core.experimental.serializer.PropertyNameSerializer;
+import com.azure.core.util.CoreUtils;
+import com.fasterxml.jackson.annotation.JsonProperty;
+import reactor.core.publisher.Mono;
+
+import java.lang.reflect.Field;
+import java.lang.reflect.Member;
+import java.lang.reflect.Method;
+
+/**
+ * Jackson based get property name implementation of the {@link PropertyNameSerializer} interface.
+ */
+public class JacksonPropertyNameSerializer implements PropertyNameSerializer {
+    @Override
+    public Mono<String> getSerializerMemberName(Member member) {
+
+        return Mono.fromCallable(() -> {
+            String propertyName = null;
```

Comment:
I believe we need to account for `JsonIgnore`.

Reply:
We provide @FieldIgnore for FieldBuilder to ignore the field for SearchIndex, which provide the flexibility of decouple the use case of FieldBuilder and serialization (Document). The `getSerializerMemberName` is useful for FieldBuilder, customer can use `serialize` and `deserialize` API for document serialization.